src/library/scala/collection/mutable/OpenHashMap.scala

```diff
@@ -64,7 +72,22 @@ extends AbstractMap[Key, Value]
   private[this] val actualInitialSize = OpenHashMap.nextPositivePowerOfTwo(initialSize)
 
   private var mask = actualInitialSize - 1
-  private var table : Array[Entry] = new Array[Entry](actualInitialSize)
+
+  /** The state of each of the slots in the hash table.
+    * One of `Empty`, `Deleted` or `Occupied`.
+    */
+  private[this] var slotStates = new Array[Byte](actualInitialSize)
```

Comment:
I didn't try to add a `Byte` to `OpenEntry`. The thought crossed my mind, since that would preserve memory locality. But the direction that I'm headed is to eliminate the overhead of objects like `OpenEntry`, to support higher capacities than a chained hash map in a JVM. Because, if this class isn't good for that, then what is it good for?

I didn't realize that `OpenEntry`'s inheritance from `HashEntry` was adding an extra word. I assumed that it wasn't adding overhead, since it wasn't adding feature. The response to that, though, is to remove that dependence.